sdgym/sdgym_result_explorer/result_explorer.py

```diff
@@ -0,0 +1,194 @@
+                Bucket=self._bucket_name, Prefix=self._prefix, Delimiter='/'
+            )
+            return [cp['Prefix'].split('/')[-2] for cp in response.get('CommonPrefixes', [])]
+
+        else:
+            return [d for d in os.listdir(self.path) if os.path.isdir(os.path.join(self.path, d))]
+
+    def _validate_access(self, results_folder_name, dataset_name, synthesizer_name, type):
+        """Validates the synthesizer access."""
+        end_filename = f'{synthesizer_name}_'
+        if type == 'synthetic_data':
+            end_filename += 'synthetic_data.csv'
+        elif type == 'synthesizer':
+            end_filename += 'synthesizer.pkl'
+
+        date = results_folder_name.split('_')[-3:]
+        date = '_'.join(date)
+        file_path = f'{results_folder_name}/{dataset_name}_{date}/{synthesizer_name}/{end_filename}'
+        if not self._is_s3_path:
+            synthesizer_path = os.path.join(
+                self.path, results_folder_name, f'{dataset_name}_{date}', synthesizer_name
+            )
+            if not os.path.exists(synthesizer_path):
+                raise ValueError(
+                    f"Synthesizer '{synthesizer_name}' for dataset '{dataset_name}' in"
+                    f" run '{results_folder_name}' does not exist."
+                )
+
+            if not os.path.isfile(os.path.join(synthesizer_path, end_filename)):
+                raise ValueError(
+                    f"Synthesizer file '{end_filename}' does not exist in '{synthesizer_path}'."
+                )
+        else:
+            try:
+                self._s3_client.head_object(
+                    Bucket=self._bucket_name, Key=f'{self._prefix}{file_path}'
+                )
+            except ClientError as e:
+                raise ValueError(
+                    f"Synthesizer '{synthesizer_name}' for dataset '{dataset_name}' "
+                    f"in run '{results_folder_name}' does not exist."
+                ) from e
+
+        return file_path
+
+    def load_synthesizer(
+        self,
+        results_folder_name='SDGym_run_04_18_2025',
+        dataset_name='adult',
+        synthesizer_name='ctgan',
+    ):
+        """Load the synthesizer object.
+
+        Args:
+            run_folder_name (str): The name of the folder for the desired run.
+            dataset_name (str): The name of the dataset.
+            synthesizer_name (str): The name of the synthesizer (eg. ctgan).
+
+        Returns:
+            sdgym.synthesizers.BaselineSynthesizer
+        """
+        file_path = self._validate_access(
+            results_folder_name, dataset_name, synthesizer_name, 'synthesizer'
+        )
+        if self._is_s3_path:
+            response = self._s3_client.get_object(
+                Bucket=self._bucket_name, Key=f'{self._prefix}{file_path}'
+            )
+            synthesizer = pickle.loads(response['Body'].read())
+        else:
+            with open(os.path.join(self.path, file_path), 'rb') as f:
```

Comment:
Would this work on Windows? Because `file_path = f'{results_folder_name}/{dataset_name}_{date}/{synthesizer_name}/{end_filename}'`, while `self.path=C:\Users\data\results` or something like that.